File: Python/1637. Widest Vertical Area Between Two Points Containing No Points.py
class Solution:
    def maxWidthOfVerticalArea(self, points: List[List[int]]) -> int:
        recoder = []
        for i in range(0, len(points)):
            recoder.append(points[i][0])
        recoder.sort()
        xmax = 0
        for i in range(0, len(recoder)-2):
            xmax = max(xmax, recoder[i+1] - recoder[i])
        return xmax

File: C++/1637. Widest Vertical Area Between Two Points Containing No Points.cpp
class Solution {
public:
    int maxWidthOfVerticalArea(vector<vector<int>>& points) {
        vector<int> temp;
        int m = 0;

        for(vector<int> i: points) temp.push_back(i[0]);

        sort(temp.begin(), temp.end());

        for(int i = 1; i < temp.size(); i++) m = max(m, temp[i] - temp[i - 1]);

        return m;
    }
};
